tests/regression/70-transform/02-deadcode.c

```diff
@@ -0,0 +1,139 @@
+// SKIP: this is an input file for cram tests
+
+#include <stdnoreturn.h>
+
+// requires sem.noreturn.dead_code
+extern noreturn void abort();
+
+
+// only called with positive n
+int basic1(int n) {
+  int a = 0, b = 1, c;
+
+  if (n < 0)
+    return 0;
+
+  for (int i = 0; i < n; i++) {
+    c = a + b;
+    a = b;
+    b = c;
+  }
+
+  return a;
+}
+
+// only called with negative n
+int basic2(int n) {
+  int a = 0;
+
+  if (n < 0)
+    return 0;
+
+  for (int i = 0; i < n; i++)
+    a += i + n;  // bug in dead code detection: is not found dead by goblint, because it is not included in the result
```

Comment:
Hmm? Is this comment outdated? Because it is removed by the transformation.

Reply:
The transformation works, but Goblint's existing dead code detection does not:

`./goblint --enable ana.int.interval --enable sem.noreturn.dead_code tests/regression/70-transform/02-deadcode.c`

```text
...
[Warning][Deadcode] Function 'basic2' has dead code:
  on line 32 (tests/regression/70-transform/02-deadcode.c:32-32)
  on line 35 (tests/regression/70-transform/02-deadcode.c:35-35)
...
```

The body of the loop is not reported as dead. This is since the body doesn't appear in the result, and currently Goblint only reports dead code for nodes that are in the result, but dead in all contexts.

Reply:
I've opened #1036 to fix this.